auction.sympl

```diff
@@ -33,95 +40,273 @@ schema Auction:
     creator: str
     closed: bool
 
-# Returns all the auctions in our storage
+# Admins are the only identities allowed to:
+# 1. Create the Premium auction channel
+# 2. Select another admin to replace themselves (one serving admin at a time)
+# 3. Add/remove members in the Premium channel
+# The very first time an identity is chosen to be an admin, that permission is given automatically.
+# Only one admin exists and has admin privileges at a time.
+schema Admin:
+    id: KeyAlias
+
+# A channel for private auctions.
+schema PremiumChannel:
+    channel_id: ChannelName
+    members : List[KeyAlias]
+
+###################################
+# Auction API and Implementations #
+###################################
+
+# Returns all auctions in storage.
 @clientside
 def get_auctions() -> List[Auction]:
-    return _get_auctions()
+    return cvm.storage.query(AuctionStatic).values()
 
-@helper
-def _get_auctions() -> List[Auction]:
-    return [row.value for row in cvm.storage.query(AuctionStatic).execute() ]
+# Returns a specific auction (if it exists).
+@clientside
+def get_auction(id: Identifier, channel: ChannelName=PUBLIC) -> Optional[Auction]:
+    return cvm.storage.get(channel, AuctionStatic, id)
 
-# Returns a specific auction (if it exists)
+# Creates an auction.
+# Defaults to the PUBLIC channel, but members of the premium channel may post to it by specifying its ChannelName.
 @clientside
-def get_auction(id: Identifier) -> Optional[Auction]:
-    return cvm.storage.get(PUBLIC, AuctionStatic, id)
+def create_auction(product_description: str, initial_price: int, channel: ChannelName=PUBLIC) -> None:
+    with PostTxArgs(channel):
```

Comment:
if the channel is a private one and the KA calling this function is not a member, you will just get a missing key error.
This channel seems to exist already, can you pull it out of storage instead of passing it as an input?

Reply:
I'd like to maintain functionality for private channel members to post auctions/bids in either channel, but you're definitely right about just pulling the channel from storage. Maybe I replace this parameter with a bool or something (like `private: bool=False`)?